### PR TITLE
Make toast

### PR DESCRIPTION
**Pull request by john12321**
**Toast**
- Add new toast messages to replace the temp success/error notifications (e.g. Form R saves/ doesn't save)
- Redo the toast notification logic so we can include more info with the error msg (e.g. error code). 
- Error toast has a link to Tech Support
- Clicking the Tech Support toast link will open email client and prepopulate with device data, current URL, http error. (`see NOTE`)

**Support page**
- clicking on LO link (not London) opens email client and pre-populates email body with their device info. (`see NOTE`)

TODO (follow-up tickets)
- Differentiate between a form save/update and submission (longstanding issue with messaging as both use the same thunk/api call (possible solution is to pass a flag to showToast)
- Include `AWS X-Amzn-Trace-Id` in emailto data (easy enough to grab from headers - just need to rework a few comps so left to a follow-up ticket)

TIS21-4652

`NOTE` - in addition to the current email header info: tisId, GMC.

### CODE REVIEW

File: components/dsp/__test__/Support.test.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+} from "../../../redux/slices/traineeProfileSlice";
+
+describe("Support", () => {
+  it("should show the tisId and GMC number plus User Agent data in the Support link href", () => {
+    jest
+      .spyOn(window.navigator, "userAgent", "get")
+      .mockReturnValue("Mocked User Agent Info");
+    jest.spyOn(window, "location", "get").mockReturnValue({
+      ...window.location,
+      href: "http://example.com"
```
Comment (ReubenRobertsHEE) [inline components/dsp/__test__/Support.test.tsx:22]: What, not evertonfc.com? :trollface:

File: styles/globals.scss [hunk 2]
```diff
@@ -4,10 +4,12 @@
 
 $nhsFont: “Frutiger W01”, Arial, sans-serif;
 $nhsBlue: #005eb8;
+$nhsSuccess: #006400;
+$nhsError: #a7171a;
```
Comment (ReubenRobertsHEE) [inline styles/globals.scss:8]: Again, do we need styling for info and warning messages, even if they are not used right now?

File: components/common/ToastMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        data-cy="toastMsgSuccess"
+        msg={message}
+        type={messageType}
+      />,
+      {
+        icon: <FontAwesomeIconWrapper messageType={messageType} />,
+        autoClose: 6000
+      }
+    );
+  } else if (messageType === ToastType.ERROR) {
```
[the remark this answers is not in the document]
Comment (john12321) [reply components/common/ToastMessage.tsx:76]: @ReubenRobertsHEE Good suggestion but where's the fun in that? 😁
I'm feeling a bit lazy, are you OK if I remove the INFO and WARNING from the enum until we need them? #agileLaziness

File: styles/globals.scss (hunk 2 above)
Comment (john12321) [reply styles/globals.scss:8]: See above comment on your comment @ReubenRobertsHEE . Second option is my favourite.

File: components/dsp/__test__/Support.test.tsx (hunk 1 above)
Comment (john12321) [reply components/dsp/__test__/Support.test.tsx:22]: I'll fix this egregious error @ReubenRobertsHEE

File: components/common/ToastMessage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ReubenRobertsHEE) [reply components/common/ToastMessage.tsx:76]: Sure, just-in-time is the right time! 💨

File: components/dsp/__test__/Support.test.tsx (hunk 1 above)
Comment (ReubenRobertsHEE) [reply components/dsp/__test__/Support.test.tsx:22]: Be sure not to let it happen again, is all.

**Comment by ReubenRobertsHEE** [comment]
@john12321 is there a test for the email that includes the Page URL and Error message sections (or is it worth having one)? I can't see any test that examines the tail end of `Page URL:%0A${getPageURL()}%0A%0AError message:%0A${actionErrorMsg}%0A%0A%0A%0A`}` from the ToastMessage.

**Comment by john12321** [comment]
> @john12321 is there a test for the email that includes the Page URL and Error message sections (or is it worth having one)? I can't see any test that examines the tail end of `Page URL:%0A${getPageURL()}%0A%0AError message:%0A${actionErrorMsg}%0A%0A%0A%0A`}` from the ToastMessage.

@ReubenRobertsHEE  I thought this was covered in the the `expectedHref` in the `ToastMessage.test.tsx` test file: 

` const expectedHref =
        "mailto:tis.support@hee.nhs.uk?subject=TSS tech support query (GMC no. 11111111, TIS ID 123)&body=Browser and OS info:%0AUser Agent: Mocked User Agent Info%0A%0APage URL:%0Ahttps://www.evertonfc.com%0A%0AError message:%0A808 State%0A%0A%0A%0A";`
        
...or am I getting the wrong end of the stick (more likely lol)?

**Comment by ReubenRobertsHEE** [comment]
No you're quite right - sorry, I missed it (and nice ref to 808 State, kudos!)

**Comment by john12321** [comment]
> No you're quite right - sorry, I missed it (and nice ref to 808 State, kudos!)

One for the yoof 😁

**Review by ReubenRobertsHEE — APPROVED** [review]

**Pull request merged**